projects/graph/src/draw.py

```diff
@@ -1,14 +1,50 @@
 """
 General drawing methods for graphs using Bokeh.
 """
-
-from bokeh.io import show, output_file
+from bokeh.io import output_notebook, output_file, show
 from bokeh.plotting import figure
-from bokeh.models import (GraphRenderer, StaticLayoutProvider, Circle, LabelSet,
-                          ColumnDataSource)
+from bokeh.models import ColumnDataSource, GraphRenderer, StaticLayoutProvider, LabelSet, Circle, MultiLine
+from bokeh.palettes import Category20c
+from bokeh.models.graphs import NodesAndLinkedEdges
+from graph import Graph
+import math
 
 
 class BokehGraph:
     """Class that takes a graph and exposes drawing methods."""
-    def __init__(self):
-        pass  # TODO
+    def __init__(self, graph):
+        self.graph=graph
+    
+    def graph_render(self):
+        nodes = list(self.graph.vertices.keys())
+        N=len(nodes)
+
+        #Create plot
+        plot = figure(title = 'Graph', x_range = (-1.5, 1.5) , y_range = (-1.5, 1.5), tools = "", toolbar_location = None)
+
+        graph = GraphRenderer()
+        
+        graph.node_renderer.glyph = Circle(radius = 0.1, fill_color = "fill_color")  
+        graph.node_renderer.data_source.data = dict(
+            index = nodes,
+            fill_color = ['blue']*N
+            )
+
+            
+        graph.edge_renderer.data_source.data = dict(start=[0]*N,end=nodes)
+
+        ##start of layout code
+        circ = [i*2*math.pi/len(nodes) for i in range(len(nodes))]
+        x = [math.cos(i) for i in circ]
+        y = [math.sin(i) for i in circ]
+
+
```

Comment:
Nice job! I like that you played around with some of the values a little - it's fun to see what happens when you change things. I'm glad you have bokeh working on your Mac for the rest of this assignment :)